### ISSUE TITLE
Reshape Error when onnx to tensorrt

### ISSUE THREAD

**Issue by selimceylan** [NVIDIA/TensorRT#1598]
## Description
Reshape error while converting onnx to TensorRT for C++ API.

ERROR CODE:

ModelImporter.cpp:769: While parsing node number 14 [Reshape -> "StatefulPartitionedCall/Postprocessor/Reshape:0"]:nModelImpoerter.cpp:770: --- Begin node --- nModelImporter.cpp 771: input: "StatefulPartitionedCall/Postprocessor/ExpandDims:0"
input: "const_fold_opt__5430"
output: "StatefulPartitionedCall/Postprocessor/Reshape:0"
name: "StatefulPartitionedCall/Postprocessor/Reshape"
op_type: "Reshape"
nModelImporter.cpp:772: --- End Node ---nModelImporter.cpp:775: ERROR: nModelImporter.cpp:162 In function parseGraph:
[6] Invalid Node - StatefulPartitionedCall/Postprocessor/Reshape
Attribute not found: allowzeronERROR: could not parse the model.

## Environment

**TensorRT Version**: 8.2.0
**NVIDIA GPU**: NVIDIA GeForce RTX 2080 TI
**NVIDIA Driver Version**:  471.96
**CUDA Version**: 10.2
**CUDNN Version**: 8.2.1
**Operating System**: Windows 10 
**Python Version (if applicable)**: 3.9
**Tensorflow Version (if applicable)**: 2.6.0
**Onnx Version: 1.10.1
 **Opencv Version: 4.5.4

## Relevant Files

I download this saved_model.pb file and converted it to onnx file with opset 14. Although, it converted onnx successfully it does not convert TensoRT.
Link of saved_model.pb: https://tfhub.dev/tensorflow/retinanet/resnet50_v1_fpn_1024x1024/1

## Steps To Reproduce

I used the code of TensorRT C++ inference API from: https://learnopencv.com/how-to-run-inference-using-tensorrt-c-api/

This is the command for convert onnx: python -m tf2onnx.convert --saved-model saved_model-file_path --output output_file_path --opset 14 --fold_const

**Comment by ttyio** [comment]
Hello @selimceylan , thanks for reporting, there is an issue in ONNX2TRT, could you export ONNX using `--opset 13` before we fix this? thanks!

**Comment by selimceylan** [comment]
Thanks for reply, when ı used the --opset 13 version (also other versions) different error occurred: 

ModelImporter.cpp:769: While parsing node number 3048 [TopK -> "StatefulPartitionedCall/Postprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/SortByField/TopKV2:0"]:nModelImporter.cpp:770: --- Begin node ---nModelImporter.cpp:771: input: "StatefulPartitionedCall/Postprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/Concatenate/concat_1:0"
input: "Unsqueeze__4519:0"
output: "StatefulPartitionedCall/Postprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/SortByField/TopKV2:0"
output: "StatefulPartitionedCall/Postprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/SortByField/TopKV2:1"
name: "StatefulPartitionedCall/Postprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/SortByField/TopKV2"
op_type: "TopK"
attribute {
  name: "sorted"
  i: 1
  type: INT
}
nModelImporter.cpp:772: --- End node ---nModelImporter.cpp:775: ERROR: builtin_op_importers.cpp:4519 In function importTopK:
[8] Assertion failed: (inputs.at(1).is_weights()) && "This version of TensorRT only supports input K as an initializer."nERROR: could not parse the model.

**Comment by ttyio** [comment]
Hmm @selimceylan , the `K` has to be static value in TRT `TopK`. 

https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_network_definition.html#a13fff8068d7794918c865d7f854a0508

Can you adjust your model to use fixed `K`? thanks!

**Comment by selimceylan** [comment]
Unfortunately, I can not change values from .pb file and this is pretrained model. For another model I encountered this:
 
[shuffleNode.cpp::nvinfer1::builder::ShuffleNode::symbolicExecute::387] Error Code 4: Internal Error (enrich_semantics/enrich_P5/Conv/required_space_to_batch_paddings/paddings_Unsqueeze__2014: IShuffleLayer applied to shape tensor must have 0 or 1 reshape dimensions: dimensions were [1,2])nModelImporter.cpp:769: While parsing node number 434 [Pad -> "Pad__2028:0"]:nModelImporter.cpp:770: --- Begin node ---nModelImporter.cpp:771: input: "build_pyramid/fuse_P5/BiasAdd:0"
input: "Concat__9859:0"
output: "Pad__2028:0"
name: "Pad__2028"
op_type: "Pad"
nModelImporter.cpp:772: --- End node ---nModelImporter.cpp:775: ERROR: ModelImporter.cpp:179 In function parseGraph:
[6] Invalid Node - Pad__2028
[shuffleNode.cpp::nvinfer1::builder::ShuffleNode::symbolicExecute::387] Error Code 4: Internal Error (enrich_semantics/enrich_P5/Conv/required_space_to_batch_paddings/paddings_Unsqueeze__2014: IShuffleLayer applied to shape tensor must have 0 or 1 reshape dimensions: dimensions were [1,2])

Is this also about dynamic shapes ?

**Comment by ttyio** [comment]
@selimceylan , yes the ND shape tensor is still under develop. Have you tried 

           polygraphy surgeon sanitize model.onnx --fold-constants --output model_folded.onnx

sometimes the ND shape tensor can be removed after this cmd.

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, thanks!
